packages/discovery-provider/src/api/v1/tracks.py

```diff
@@ -463,30 +463,35 @@ def get(self, track_id):
         This endpoint accepts the Range header for streaming.
         https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests
         """
-        args = stream_parser.parse_args()
-        stream_preview = args.get("preview")
+        request_args = stream_parser.parse_args()
+        is_preview = request_args.get("preview")
+        filename = request_args.get("filename")
         decoded_id = decode_with_abort(track_id, ns)
         info = get_track_stream_info(decoded_id)
 
         track = info.get("track")
         cid = track.get("track_cid")
-        if stream_preview:
+        if is_preview:
             cid = track.get("preview_cid")
+        elif filename:
+            cid = track.get("orig_file_cid")
```

Comment:
this logic is incorrect. fixing in another PR.

Reply:
I think we'll need to fix before merging since we deploy DN every day?

Reply:
yeah everything will be on feature branches until everything works.